### PR TITLE
WT-6654 Clean up test_backup15.py

### PR DESCRIPTION
**Pull request by sueloverso**
@ddanderson I was able to reproduce after about 800 iterations. After making a few minor test improvements I have not been able to reproduce after 4000 iterations. There was a path in the test that would have compared the main table against itself (and expect them to be different). I fixed that, but added an assertion at the top to make sure we weren't taking that path. So I haven't "fixed" anything. There was also a line that was repeated. We may still see this again but I cannot reproduce at the moment and these changes improve the test.

### CODE REVIEW

**Review by ddanderson — APPROVED** [review]
LGTM.  Yes, this looks like some nice cleanup that will not change the functionality of the test.  I have to say it's a bit hard to follow the logic of this test.
While you're in there, can you fix a spelling error 'bakcup' in comment at line 110?  I can't comment on it, cause it's outside the range of your changes.  Makes me think of baking cupcakes.

**Comment by sueloverso** [comment]
@ddanderson please take another look as I implemented many of the items you mentioned on slack for improvement. The `while True` statements cannot be changed but I added a comment (originally in `test_backup13.py`) stating why.

File: test/suite/test_backup15.py [hunk 1]
```diff
@@ -278,38 +281,12 @@ def add_data_validate_backups(self):
...
-
-    def test_backup15(self):
-        os.mkdir(self.bkp_home)
-        self.home = self.bkp_home
-        self.session.create(self.uri, "key_format=S,value_format=S")
-
-        self.setup_directories()
-
-        self.pr('*** Add data, checkpoint, take backups and validate ***')
-        self.add_data_validate_backups()
```
Comment (ddanderson) [inline test/suite/test_backup15.py:312]: The original code does `take_full_backup` and `take_incr_backup` in a different order every other time.  Don't know if that makes for a better test?

File: test/suite/test_backup15.py [hunk 2]
```diff
@@ -126,13 +143,17 @@ def take_full_backup(self):
 
                 shutil.copy(copy_from, copy_to)
         self.assertEqual(ret, wiredtiger.WT_NOTFOUND)
-        cursor.close()
+        bkup_c.close()
 
     def take_incr_backup(self):
         # Open the backup data source for incremental backup.
         buf = 'incremental=(src_id="ID' +  str(self.counter-1) + '",this_id="ID' + str(self.counter) + '")'
```
Comment (ddanderson) [inline test/suite/test_backup15.py:150]: The previous commit added this statement here:
```
 self.assertTrue(self.counter > 0)
```
I think that's helpful to have.  Alternatively it could be put at the top of `range_copy`, where `self.counter` is used.

**Review by ddanderson — APPROVED** [review]
Thanks, this test is much easier to understand.  I noted a couple items to consider.

File: test/suite/test_backup15.py (hunk 2 above)
Comment (sueloverso) [reply test/suite/test_backup15.py:150]: Fixed. Thanks for catching that I agree it is helpful to have.
